pkg/log/prettylogging.go

```diff
@@ -0,0 +1,164 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package log
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	maxLenLogger = 20
+	maxLenCaller = 25
+)
+
+var verbosityLevel = 0
+
+// AddFlags register command line options for pretty logging. Should be called before InitLogging.
+func AddFlags(flagset *flag.FlagSet) {
+	if flagset == nil {
+		flagset = flag.CommandLine
+	}
+
+	flagset.IntVar(&verbosityLevel, "v", verbosityLevel,
+		"Number for the log level verbosity (higher is more verbose)")
+
+	// avoid runtime error when klog's alsologtostderr option is enabled for the container
+	// this is the default in most of the submariner container command.
+	flagset.Bool("alsologtostderr", false, "Deprecated (always logging to stderr")
+}
+
+// InitLogging Initializes a new pretty logger as the concrete logr.Logger implementation in use.
+func InitLogging() {
+	zerolog.TimeFieldFormat = zerolog.TimeFormatUnixMs
+	prettyZeroLogger := createPrettyLogger()
+	logAdapter := newAdapter(&prettyZeroLogger, verbosityLevel)
+	logf.SetLogger(logAdapter)
+}
+
+func createPrettyLogger() zerolog.Logger {
+	consoleWriter := zerolog.ConsoleWriter{Out: os.Stderr, TimeFormat: time.RFC3339}
+	consoleWriter.FormatCaller = formatCaller
+	logger := log.Output(consoleWriter).With().Caller().Logger()
+
+	return logger
+}
+
+func newAdapter(zeroLogger *zerolog.Logger, maxVerbosityLevel int) logr.Logger {
+	return zeroLogContext{
+		zLogger:          zeroLogger,
+		prefix:           "",
+		currentVerbosity: 0,
+		maxVerbosity:     maxVerbosityLevel,
+	}
+}
+
+func formatCaller(i interface{}) string {
+	return truncate(i, maxLenCaller)
+}
+
+type zeroLogContext struct {
+	zLogger          *zerolog.Logger
+	prefix           string
+	currentVerbosity int
+	maxVerbosity     int
+}
+
+func (ctx zeroLogContext) clone() zeroLogContext {
```

Comment:
Same with the other method receivers.

```suggestion
func (ctx *zeroLogContext) clone() zeroLogContext {
```